lib/app/timer.ex

```diff
@@ -63,6 +65,278 @@ defmodule App.Timer do
     |> Repo.update()
   end
 
+  @doc """
+  Updates a timer object.
+
+  ## Examples
+
+      iex> update_timer(%{id: 1, start: ~N[2022-07-11 05:15:31], stop: ~N[2022-07-11 05:15:37]})
+      {:ok, %Timer{id: 1, start: ~N[2022-07-11 05:15:31], stop: ~N[2022-07-11 05:15:37}}
+
+  """
+  def update_timer(attrs \\ %{}) do
+    get_timer!(attrs.id)
+    |> changeset(attrs)
+    |> Repo.update()
+  end
+
+  @doc """
+  Updates a timer object inside a list with timer changesets.
+  This function is only useful for form validations, since it replaces the errored changeset
+  according to the index that is passed, alongside the list and the fields to update the timer.
+
+  It returns {:ok, []} in case the update is successful.
+  Otherwise, it returns {:error, updated_list}, where `error_term` is the error that occurred and `updated_list` being the updated item changeset list with the error.
+  """
+  def update_timer_inside_changeset_list(
+        timer_id,
```

Comment:
could the function signature here be a Map/Struct with these fields? 💭

Reply:
Just pushed a commit with this in mind. I didn't put *everything* into a struct but all the timer-related ones. The index and changeset list stayed the same 😄